### PR TITLE
fixes #4274 - remove I18n deprecation warning

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

**Comment by isratrade** [comment]
Here's a good summary of the deprecation warning.  
http://stackoverflow.com/questions/20361428/rails-i18n-validation-deprecation-warning

File: config/application.rb [hunk 1]
```diff
@@ -88,6 +88,7 @@ class Application < Rails::Application
     # config.time_zone = 'Central Time (US & Canada)'
     config.time_zone = 'UTC'
 
+    I18n.enforce_available_locales = false
```
Comment (isratrade) [inline config/application.rb]: The previous default before Rails 3.2.14 was `false`, so I'm keeping it as `false`.  We can change it to `true` but didn't see where we define available locales?  @lzap

Comment (isratrade) [inline config/application.rb]: I see now where available locales are defined.  I'll switch to `true`
`https://github.com/theforeman/foreman/blob/develop/lib/foreman/gettext/support.rb`

**Comment by isratrade** [comment]
seems to be Jenkins failure only for 1.9.3 on sqlite.  will re [test]

**Comment by domcleal** [comment]
Not tested, but will this cause any sort of error if a user requests a locale via say a browser Accept header that we don't support?

**Comment by lzap** [comment]
Yeah we need definitely to test this scenario and via the API as well. We have been there...

**Comment by domcleal** [comment]
Interestingly, this breaks the rpmbuild which precompiles assets:

<pre>
+ /usr/bin/ruby193-rake assets:precompile:all RAILS_ENV=production --trace
rake aborted!
undefined method `enforce_available_locales=' for I18n:Module
/builddir/build/BUILD/foreman-1.5.0/config/application.rb:91:in `<class:Application>'
/builddir/build/BUILD/foreman-1.5.0/config/application.rb:45:in `<module:Foreman>'
/builddir/build/BUILD/foreman-1.5.0/config/application.rb:44:in `<top (required)>'
/opt/rh/ruby193/root/usr/share/rubygems/rubygems/custom_require.rb:36:in `require'
/opt/rh/ruby193/root/usr/share/rubygems/rubygems/custom_require.rb:36:in `require'
/builddir/build/BUILD/foreman-1.5.0/Rakefile:1:in `<top (required)>'
/opt/rh/ruby193/root/usr/share/ruby/rake/rake_module.rb:25:in `load'
/opt/rh/ruby193/root/usr/share/ruby/rake/rake_module.rb:25:in `load_rakefile'
/opt/rh/ruby193/root/usr/share/ruby/rake/application.rb:501:in `raw_load_rakefile'
/opt/rh/ruby193/root/usr/share/ruby/rake/application.rb:82:in `block in load_rakefile'
/opt/rh/ruby193/root/usr/share/ruby/rake/application.rb:133:in `standard_exception_handling'
/opt/rh/ruby193/root/usr/share/ruby/rake/application.rb:81:in `load_rakefile'
/opt/rh/ruby193/root/usr/share/ruby/rake/application.rb:65:in `block in run'
/opt/rh/ruby193/root/usr/share/ruby/rake/application.rb:133:in `standard_exception_handling'
/opt/rh/ruby193/root/usr/share/ruby/rake/application.rb:63:in `run'
/opt/rh/ruby193/root/usr/bin/rake:32:in `<main>'
</pre>

**Comment by lzap** [comment]
Btw is this being reported on the IRC channel at least?

**Comment by isratrade** [comment]
@lzap, I didn't see this issue on IRC.  It just bothered me when I was running tests.  
< 3.2.14   `I18n.enforce_available_locales = false`
=> 3.2.14 `I18n.enforce_available_locales = true` with deprecation warning
Since we package using 3.2.8 (I think), I assume it's best to set it to `false` for now.  ack?

**Comment by lzap** [comment]
Yeah false for now.

I really do not understand why they default to validate this. I share Dom's concern about what happens if user with strange browser locale (correct but unsupported) visits the page. Then an exception page? That is too much.

I think the default behavior is okay - we fallback to English if anything goes wrong. We use FastGettext anyway and we have different detection of available locales based on listing MO files in locale/*\* directory. So I am fine with _false_ even for future Rails 4.0+ versions.

**Comment by lzap** [comment]
Remember - we keep Rails i18n "yaml" files and our gettext PO/MO files in sync. That means once we add a language, we also fetch/update those "yaml" files. We are fine.

**Comment by isratrade** [comment]
re-committed to `false`

**Comment by domcleal** [comment]
We're still seeing the RPM build failure while precompiling assets mentioned in https://github.com/theforeman/foreman/pull/1216#issuecomment-34327666, any idea why this might be?

**Comment by isratrade** [comment]
I changed the syntax but I thought both should work.

``` ruby
-    I18n.enforce_available_locales = false
+   config.i18n.enforce_available_locales = false
```

**Comment by isratrade** [comment]
@domcleal, not sure why this also failed

**Comment by domcleal** [comment]
@isratrade same random failures mostly, but also the rpmbuild is still failing.  Same issue as above, but further down the stack since the config/i18n change...

<pre>
+ /usr/bin/ruby193-rake assets:precompile:all RAILS_ENV=production --trace
** Invoke assets:precompile:all (first_time)
** Execute assets:precompile:all
** Invoke assets:precompile:primary (first_time)
** Invoke assets:environment (first_time)
** Execute assets:environment
** Invoke environment (first_time)
** Execute environment
rake aborted!
undefined method `enforce_available_locales=' for I18n:Module
/opt/rh/ruby193/root/usr/share/gems/gems/activesupport-3.2.8/lib/active_support/i18n_railtie.rb:60:in `block in initialize_i18n'
/opt/rh/ruby193/root/usr/share/gems/gems/activesupport-3.2.8/lib/active_support/i18n_railtie.rb:53:in `each'
/opt/rh/ruby193/root/usr/share/gems/gems/activesupport-3.2.8/lib/active_support/i18n_railtie.rb:53:in `initialize_i18n'
/opt/rh/ruby193/root/usr/share/gems/gems/activesupport-3.2.8/lib/active_support/i18n_railtie.rb:34:in `block in <class:Railtie>'
/opt/rh/ruby193/root/usr/share/gems/gems/activesupport-3.2.8/lib/active_support/lazy_load_hooks.rb:34:in `call'
/opt/rh/ruby193/root/usr/share/gems/gems/activesupport-3.2.8/lib/active_support/lazy_load_hooks.rb:34:in `execute_hook'
/opt/rh/ruby193/root/usr/share/gems/gems/activesupport-3.2.8/lib/active_support/lazy_load_hooks.rb:43:in `block in run_load_hooks'
/opt/rh/ruby193/root/usr/share/gems/gems/activesupport-3.2.8/lib/active_support/lazy_load_hooks.rb:42:in `each'
/opt/rh/ruby193/root/usr/share/gems/gems/activesupport-3.2.8/lib/active_support/lazy_load_hooks.rb:42:in `run_load_hooks'
/opt/rh/ruby193/root/usr/share/gems/gems/railties-3.2.8/lib/rails/application/finisher.rb:59:in `block in <module:Finisher>'
/opt/rh/ruby193/root/usr/share/gems/gems/railties-3.2.8/lib/rails/initializable.rb:30:in `instance_exec'
/opt/rh/ruby193/root/usr/share/gems/gems/railties-3.2.8/lib/rails/initializable.rb:30:in `run'
/opt/rh/ruby193/root/usr/share/gems/gems/railties-3.2.8/lib/rails/initializable.rb:55:in `block in run_initializers'
/opt/rh/ruby193/root/usr/share/gems/gems/railties-3.2.8/lib/rails/initializable.rb:54:in `each'
/opt/rh/ruby193/root/usr/share/gems/gems/railties-3.2.8/lib/rails/initializable.rb:54:in `run_initializers'
/opt/rh/ruby193/root/usr/share/gems/gems/railties-3.2.8/lib/rails/application.rb:136:in `initialize!'
/opt/rh/ruby193/root/usr/share/gems/gems/railties-3.2.8/lib/rails/railtie/configurable.rb:30:in `method_missing'
/builddir/build/BUILD/foreman-1.5.0/config/environment.rb:5:in `<top (required)>'
</pre>

**Comment by isratrade** [comment]
I uncommented `config.i18n.default_locale = :en` since I read that 

``` ruby
    config.i18n.enforce_available_locales = false
```

needs to be defined before `config.i18n.default_locale =`

@domcleal, there are still failures of 2 of 9 matrix, but I think it's something else.

**Comment by domcleal** [comment]
Sorry to say it's still the same error about the I18n module.  We might need to go into more depth and find out if there are some version differences in the RPMs, or maybe with Rails itself?  (Remember the RPMs are 3.2.8.)  @lzap do you have time?

**Comment by isratrade** [comment]
@domcleal, I'll just enjoy seeing deprecation error when running tests in the meantime.

**Comment by domcleal** [comment]
@isratrade don't let it get to you :)

**Comment by isratrade** [comment]
I re-based and moved the syntax back to `config.i18n.enforce_available_locales = false` and the tests pass now.

**Comment by domcleal** [comment]
That's only because RPMs are no longer built during PR tests, the problem still needs investigating and fixing.

**Comment by isratrade** [comment]
@lzap, closing this stale PR.

**Pull request closed without merging**